File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stk = []
        for char in s:
            if char=='(':
                stk.append(')')
            elif char=='[':
                stk.append(']')
            elif char=='{':
                stk.append('}')
            elif len(stk)>0 and char==stk[len(stk)-1]:
                stk.pop()
            else: return False
        return len(stk) == 0

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stk = new Stack<>();
        for(char c: s.toCharArray()){
            if(c=='(') stk.push(')');
            else if(c=='[') stk.push(']');
            else if(c=='{') stk.push('}');
            else if(!stk.isEmpty() && stk.peek()==c) stk.pop();
            else return false;
        }
        return stk.isEmpty();
    }
}